Bootcamp2015-AmazingRace/Views/MainPage.xaml.cs

```diff
@@ -30,5 +36,25 @@ protected override void OnNavigatedTo(NavigationEventArgs e)
             // If you are using the NavigationHelper provided by some templates,
             // this event is handled for you.
         }
+
+        private async void GoogleLoginClick(object sender, Windows.UI.Xaml.RoutedEventArgs e)
+        {
+            var result = await App.Api._mobileServiceClient.LoginAsync(MobileServiceAuthenticationProvider.Google);
+        }
+
+        public void ContinueWebAuthentication(Windows.ApplicationModel.Activation.WebAuthenticationBrokerContinuationEventArgs args)
+        {
+            if (args.Kind == ActivationKind.WebAuthenticationBrokerContinuation)
+            {
+                App.Api._mobileServiceClient.LoginComplete(args as WebAuthenticationBrokerContinuationEventArgs);
+            }
+            LoginPrompt.Text = "Thank you for logging in. Please press Continue to proceed.";
+            GoogleLogin.Visibility = Windows.UI.Xaml.Visibility.Collapsed;
+        }
```

Comment:
This logic works better in a ViewModel because you're not tied to the View's specific implementation. Here you are assigning the Text property of the visual element, where Binding would shield you from this.

The GoogleLogin visibility could be bound to a boolean property in the ViewModel (using a BooleanToVisibilityConverter in XAML) which may make more contextual sense. eg

```
public void CompleteLoginProcess(WebAuthenticationBrokerContinuationEventArgs args)
{
     _api.LoginComplete(args);
     LoginPrompt = "Thank you...";
     IsLoginEnabled = false;
     IsContinueEnabled = true;
}
```